Create ui-c8y-1022-14-2-disable-inventory-roles-select-if-sso-user-is-not-able-to.md (#3622)
SoftwareAG/c8y-docs: content/change-logs/platform-services/ui-c8y-1022-14-2-disable-inventory-roles-select-if-sso-user-is-not-able-to.md

---
date: ""

> title: Disable inventory roles select if SSO user is not able to change them.

Reviewer: ```suggestion
title: Disabled inventory roles selection if the SSO user cannot change them
```

Reviewer: @jdre-c8y Is there a difference between "is not able" and "does not have permission"? Or is the user not able because they lack permission, as I rephrased it?

Reviewer 2: it's not about lack of permissions. like I said above, this user is managed by external SSO server and we just can't change it here (in Administration app), no matter what permissions we have.
also it's not in "SSO access mapping configuration"- this dropdown is in user details view.

product_area: Platform services
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-0UgqXH1Ys
    label: Administration
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-63881
version: 1022.14.2
---

> When for SSO access mapping configuration option "Roles selected in the rules below will be reassigned to a user on each log in and other ones will be cleared is selected", any inventory roles mapping change for external user could be change in dropdown list, but could not be saved, which might be misleading. Now, if user is not able to change inventory roles mapping, dropdown is disabled.

Reviewer 2: ```suggestion
When for SSO access mapping configuration option "Roles selected in the rules below will be reassigned to a user on each log in and other ones will be cleared" is selected, any inventory roles mapping change for external user could be change in dropdown list, but could not be saved, which might be misleading. Now, if user is not able to change inventory roles mapping, dropdown is disabled.
```

> When for SSO access mapping configuration option "Roles selected in the rules below will be reassigned to a user on each log in and other ones will be cleared" is selected, any inventory roles mapping change for external user could be change in dropdown list, but could not be saved, which might be misleading. Now, if user is not able to change inventory roles mapping, dropdown is disabled.

Reviewer: ```suggestion
In the SSO access mapping configuration, if the option "Roles selected in the rules below will be reassigned to a user on each login and other ones will be cleared" was selected, the inventory roles mapping for an external user could be changed in the dropdown list, but could not be saved, which was misleading. Now, if the user does not have the permission to change the inventory roles mapping, the dropdown is disabled.
```

Reviewer: @jdre-c8y What do you mean by "an external user"?

Reviewer 2: it's kind of user that is not managed by cumulocity platform but by external SSO server. this kind of user cannot be edited in Administration app and he logs in by external sso server provided login page, not our regular cumulocity login page.

> In the SSO access mapping configuration, if the option "Roles selected in the rules below will be reassigned to a user on each login and other ones will be cleared" was selected, the inventory roles mapping for an external user could be changed in the dropdown list, but could not be saved, which was misleading. Now, if the user does not have the permission to change the inventory roles mapping, the dropdown is disabled.

Reviewer: ```suggestion
In the user details, if for the SSO access mapping the option "Roles selected in the rules below will be reassigned to a user on each login and other ones will be cleared" was selected, the inventory roles mapping for an external user could be changed in the dropdown list, but could not be saved, which was misleading. Now, if a user is not able to change the inventory roles mapping, the dropdown is disabled.
```
